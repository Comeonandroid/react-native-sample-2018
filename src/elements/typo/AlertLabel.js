import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import CustomIcon from '../../utils/icon'
import { Text, View } from '../index'


const AlertLabel = ({ children, ...rest }) => (
  <Wrapper>
    <Icon name='info' size={25} />
    <Element {...rest}>
      {children}
    </Element>
  </Wrapper>
)

AlertLabel.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  margin: 5px 0px;
  font-family: Lato;
  font-weight: 400;
  font-size: 12px;
`

const Wrapper = styled(View)`
  flex-direction: row;
`

const Icon = styled(CustomIcon) `
  margin-top: 8px;
  margin-right: 6px;
`

export default AlertLabel