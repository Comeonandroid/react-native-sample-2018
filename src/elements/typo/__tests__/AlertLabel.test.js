import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import AlertLabel from '../AlertLabel'

describe('<AlertLabel />', () => {
  itBehavesLike('aThemedComponent', { subject: <AlertLabel>Test</AlertLabel> })
})
