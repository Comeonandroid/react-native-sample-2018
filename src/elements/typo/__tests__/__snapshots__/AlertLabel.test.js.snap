// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertLabel /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "marginRight": 6,
              "marginTop": 8,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "fontFamily": "Lato",
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`<AlertLabel /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "marginRight": 6,
              "marginTop": 8,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "fontFamily": "Lato",
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;
